analyze/test/analyze.test.ts

```diff
@@ -0,0 +1,256 @@
+import assert from "node:assert/strict";
+import test from "node:test";
+import {
+  detectBot,
+  detectSensitiveInfo,
+  generateFingerprint,
+  isValidEmail,
+} from "../index.js";
+
+const exampleContext = { characteristics: [], log: console };
+const exampleEmailOptions = {
+  allowDomainLiteral: false,
+  allow: [],
+  requireTopLevelDomain: true,
+};
+
+test("detectBot", async function (t) {
+  await t.test("should fail w/o user agent", async function () {
+    await assert.rejects(
+      detectBot(
+        exampleContext,
+        { ip: "127.0.0.1" },
+        {
+          tag: "allowed-bot-config",
+          val: { entities: [], skipCustomDetect: false },
+        },
+      ),
+      /user-agent header is empty/,
+    );
+  });
+
+  await t.test("should allow a non-bot", async function () {
+    const result = await detectBot(
+      exampleContext,
+      { ip: "127.0.0.1", headers: { "user-agent": "Mozilla/5.0" } },
+      {
+        tag: "allowed-bot-config",
+        val: { entities: [], skipCustomDetect: false },
+      },
+    );
+
+    assert.deepEqual(result, {
+      allowed: [],
+      denied: [],
+      spoofed: false,
+      verified: false,
+    });
+  });
+
+  await t.test("should detect a bot (curl)", async function () {
+    const result = await detectBot(
+      exampleContext,
+      { ip: "127.0.0.1", headers: { "user-agent": "curl/8.1.2" } },
+      {
+        tag: "allowed-bot-config",
+        val: { entities: [], skipCustomDetect: false },
+      },
+    );
+
+    assert.deepEqual(result, {
+      allowed: [],
+      denied: ["CURL"],
+      spoofed: false,
+      verified: false,
+    });
+  });
+
+  await t.test("should detect a bot (googlebot)", async function () {
+    const result = await detectBot(
+      exampleContext,
+      { ip: "127.0.0.1", headers: { "user-agent": "Googlebot/2.0" } },
+      {
+        tag: "allowed-bot-config",
+        val: { entities: [], skipCustomDetect: false },
+      },
+    );
+
+    assert.deepEqual(result, {
+      allowed: [],
+      denied: ["GOOGLE_CRAWLER"],
+      spoofed: false,
+      verified: false,
+    });
+  });
+});
+
+test("detectSensitiveInfo", async function (t) {
+  await t.test("should detect sensitive info", async function () {
+    const result = await detectSensitiveInfo(
+      exampleContext,
+      "a b@c.d e",
+      { tag: "allow", val: [] },
+      1,
+    );
+
+    assert.deepEqual(result, {
+      allowed: [],
+      denied: [{ end: 7, identifiedType: { tag: "email" }, start: 2 }],
+    });
+  });
+
+  await t.test("should not detect regular stuff", async function () {
```

Comment:
```suggestion
  await t.test("should not detect non-sensitive info", async function () {
```
As an uninformed reader of these tests, what is "regular stuff"?

Reply:
done